_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -1549,7 +1550,10 @@ mfxStatus VAAPIEncoder::CreateAuxilliaryDevice(
 
     m_caps.ddi_caps.UserMaxFrameSizeSupport = 1;
     m_caps.ddi_caps.MBBRCSupport            = 1;
-    m_caps.ddi_caps.MbQpDataSupport         = 1;
+    // Zero VAConfigAttribQPBlockSize means MBQP is not supported.
+    // VA_ATTRIB_NOT_SUPPORTED is returned by legacy drivers, treat this as supported.
+    m_caps.ddi_caps.MbQpDataSupport         =
```

Comment:
emm. So, driver explicitly returns "unsupported" and we treat this as supported. Does not mean any sense to me:). What are "legacy" drivers behaving in this way?

Reply:
@dvrogozh This was added in [[Encode] Set VAConfigAttribQPBlockSize VA attribute to report MBQP AVC support](https://github.com/intel/media-driver/commit/f403839f357c89897b309475b1252adc5cc38fbd).

I treat `VA_ATTRIB_NOT_SUPPORTED`  as "the attribute is not supported", in this case we are treating it optimistically and hope that functionality is supported (that is true for AVC VME). If the attribute is supported, we are reading a value and treat zero as "the functionality is not supported".

Sure, this may be strange, but disabling AVC VME MBQP for "new" MediaSDK working with "old" driver may be even worse.

Should we just forget about "new MediaSDK + old driver" combination? The support is cheap.